nixos/modules/system/boot/initrd-network.nix

```diff
@@ -144,13 +144,15 @@ in
         # Acquire DHCP leases.
         for iface in ${dhcpIfShellExpr}; do
           echo "acquiring IP address via DHCP on $iface..."
-          udhcpc --quit --now -i $iface -O staticroutes --script ${udhcpcScript} ${udhcpcArgs}
+          # run DHCP in the background until it acquires an address
+          udhcpc --quit -i $iface -O staticroutes --script ${udhcpcScript} ${udhcpcArgs} &
         done
       ''
 
       + cfg.postCommands));
 
     boot.initrd.postMountCommands = mkIf (cfg.flushBeforeStage2 && !config.boot.initrd.systemd.enable) ''
+      pkill udhcpc
```

Comment:
```suggestion
          # run DHCP in the background until it acquires an address
          udhcpc --quit -i $iface -O staticroutes --script ${udhcpcScript} ${udhcpcArgs} &
          udhcpc_pid="$!" # Collect PID
        done
      ''

      + cfg.postCommands));

    boot.initrd.postMountCommands = mkIf (cfg.flushBeforeStage2 && !config.boot.initrd.systemd.enable) ''
      kill -s SIGTERM $udhcpc_pid
```

Reply:
Thanks for the research and thorough comment!

Yes, I agree we would need something like this:
```
pids=()
for i in {1..5}; do
    sleep 60 &
    pids+=($!)
done

kill "${pids[@]}"
```

I was travelling but am back now. I will try it out and see if it works this weekend.

Reply:
I finally had some time to test it. It turns out that the boot process must kill any processes that are still running as I didn't need any kill statement at all. When the normal system finished booting, there was no udhcpc process.